qemu/tests/virtio_console.py

```diff
@@ -1805,6 +1807,14 @@ def test_failed_boot():
         :param cfg: virtio_console_params - Expected error message.
         """
         exp_error_message = params['virtio_console_params']
+        version_params = params["version_params"]
+        qemu_binary = utils_misc.get_qemu_binary(params)
+        output = str(process.run(qemu_binary + " --version"))
+        re_comp = re.compile(r'\d+.\d+.\d+-\d+..+')
+        output_list = re_comp.findall(output)
+        if not re.search(version_params, output_list[0]):
+            params["extra_params"] = params.get("extra_params_rhev")
+            exp_error_message = params.get("virtio_console_params_rhev")
```

Comment:
I think it's better to set new parameter original value, like this:
extra_params = " -device virtio-serial-pci,max_ports=512"
virtio_console_params = "maximum ports supported: 511"
if qemu version is 1.x.x
    extra_params_old = 
    virtio_console_params_old =
Because we should use new parameters in most cases.